rpc/namespaces/eth/api.go

```diff
@@ -513,17 +580,29 @@ func (api *PublicEthereumAPI) doCall(
 		toAddr = sdk.AccAddress(args.To.Bytes())
 	}
 
+	var msgs []sdk.Msg
 	// Create new call message
 	msg := evmtypes.NewMsgEthermint(0, &toAddr, sdk.NewIntFromBigInt(value), gas,
 		sdk.NewIntFromBigInt(gasPrice), data, sdk.AccAddress(addr.Bytes()))
+	msgs = append(msgs, msg)
 
-	if err := msg.ValidateBasic(); err != nil {
-		return nil, err
+	// convert the pending transactions into ethermint msgs
+	if blockNum == rpctypes.PendingBlockNumber {
```

Comment:
Why is the pending transaction put in here? My understanding is that only the execution result of one transaction is needed here